File: Data Structures & Algorithms/majority-element/submission-0.go
func majorityElement(nums []int) int {
    m := make(map[int]int)
    n := len(nums)

    for _, num := range nums{
        m[num]++
    }

    for i, j := range m{
        if j > n/2{
            return i
        }
    }

    return 0
}

File: Data Structures & Algorithms/majority-element-ii/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n = len(nums)
        seen = {}

        for num in nums:
            if num in seen:
                seen[num] += 1
            else:
                seen[num] = 1
        
        res= []
        for ele in seen:
            if seen[ele] > n/3:
                res.append(ele)
        
        return res 